test/Microsoft.Extensions.Logging.AzureWebAppDiagnostics.Test/AzureBlobSinkTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging.AzureWebAppDiagnostics.Internal;
+using Microsoft.WindowsAzure.Storage;
+using Moq;
+using Serilog;
+using Serilog.Core;
+using Serilog.Events;
+using Serilog.Formatting.Display;
+using Serilog.Parsing;
+using Xunit;
+
+namespace Microsoft.Extensions.Logging.AzureWebApps.Test
+{
+    public class AzureBlobSinkTests
+    {
+        [Fact]
+        public async Task WritesMessagesInBatches()
+        {
+            var blob = new Mock<ICloudAppendBlob>();
+            var buffers = new List<byte[]>();
+            blob.Setup(b => b.OpenWriteAsync()).Returns(() => Task.FromResult((Stream)new TestMemoryStream(buffers)));
+
+            var logger = CreateLogger(name => blob.Object);
+
+            for (int i = 0; i < 5; i++)
+            {
+                logger.Information("Text " + i);
+            }
+            await Task.Delay(TimeSpan.FromSeconds(1));
+
+#if NET451
+            Assert.Equal(1, buffers.Count);
+            Assert.Equal(Encoding.UTF8.GetString(buffers[0]), @"Information Text 0
+Information Text 1
+Information Text 2
+Information Text 3
+Information Text 4
+");
+#else
+            // PeriodicBatchingSink always writes first message as seperate batch on coreclr
+            Assert.Equal(2, buffers.Count);
+            Assert.Equal(Encoding.UTF8.GetString(buffers[0]), "Information Text 0\r\n");
+            Assert.Equal(Encoding.UTF8.GetString(buffers[1]), @"Information Text 1
+Information Text 2
+Information Text 3
+Information Text 4
+");
+#endif
+        }
+
+        [Fact]
+        public async Task GroupsByHour()
+        {
+            var blob = new Mock<ICloudAppendBlob>();
+            var buffers = new List<byte[]>();
+            var names = new List<string>();
+
+            blob.Setup(b => b.OpenWriteAsync()).Returns(() => Task.FromResult((Stream)new TestMemoryStream(buffers)));
+
+            var logger = CreateLogger(name =>
+            {
+                names.Add(name);
+                return blob.Object;
+            });
+
+            for (int i = 0; i < 3; i++)
+            {
+                logger.Write(new LogEvent(
+                    new DateTimeOffset(new DateTime(2016, 8, 29, i, 0, 0)),
```

Comment:
If you want to make this test cover everything, you should check also the boundaries outside the interval. Make sure that right after this, it's a new day and before is the same